### PR TITLE
[WIP] allow functions that return multiple variables in fabricate; add new reveal_outcomes and potential_outcomes functions

### PR DESCRIPTION
**Pull request by acoppock**
see https://github.com/DeclareDesign/DeclareDesign/pull/458

### CODE REVIEW

File: R/add_level.R [hunk 1]
```diff
@@ -42,21 +42,33 @@ add_top_level_internal <- function(N = NULL, ID_label = NULL,
...
-    working_data_list[[i]] <- expand_or_error(eval_tidy(
+    tmp <- expand_or_error(eval_tidy(
       data_arguments[[i]],
       append(working_data_list, list(N = N))
     ), N, i, data_arguments[[i]])
 
+    if(names(data_arguments)[i] != "") {
+      working_data_list[[names(data_arguments)[i]]] <- tmp
+    } else {
+      working_data_list <- c(working_data_list, tmp)
```
Comment (nfultz) [inline R/add_level.R:60]: Here you will need to loop over tmp explicitly and copy the items over one by one. 

Try

> c(list(a=1), list(a=2,b=3))

to see why.

**Review by nfultz — COMMENTED** [review]
I'm generally fine with this but it needs some tests.

**Comment by nfultz** [comment]
Multiple columns already works for some cases 

    df <- fabricate(N=100, Y=MASS::mvrnorm(N, 0:2, Sigma = matrix(1, 3,3)))

File: DESCRIPTION [hunk 2]
```diff
@@ -15,12 +15,13 @@ Description: Helps you imagine your data before you collect it. Hierarchical dat
 URL: https://declaredesign.org/r/fabricatr, https://github.com/DeclareDesign/fabricatr
 BugReports: https://github.com/DeclareDesign/fabricatr/issues
 Depends: R (>= 3.5.0)
-Imports: rlang (>= 0.2.0)
+Imports: rlang (>= 0.2.0), tibble
```
Comment (nfultz) [inline DESCRIPTION]: Is tibble really required (not suggests)?

Comment (acoppock) [reply DESCRIPTION]: Sorry missed this, will remove the dependency in the next maintenance release!

**Pull request merged**